packages/pwa-kit-dev/src/configs/webpack/config.js

```diff
@@ -64,9 +65,30 @@ const entryPointExists = (segments) => {
     return false
 }
 
-const findInProjectThenSDK = (pkg) => {
+/**
+ * Find the path to a dependency
+ * First check the project's node_modules, then the SDK's node_modules, then the monorepo's node_modules
+ * @param {string} pkg - name of the dependency
+ * @returns {string} path to the dependency
+ */
+const findDependency = (pkg) => {
     const projectPath = resolve(projectDir, 'node_modules', pkg)
-    return fs.existsSync(projectPath) ? projectPath : resolve(sdkDir, 'node_modules', pkg)
+    if (fs.existsSync(projectPath)) {
+        return projectPath
+    }
+    const sdkPath = resolve(sdkDir, 'node_modules', pkg)
+    if (fs.existsSync(sdkPath)) {
+        return sdkPath
+    }
+    if (isMonoRepo) {
+        // This logic assumes that the mono-repository packages are one level above the monorepo root
+        // This should be the case for lerna and nx monorepos with the default settings
+        const monoRepoPath = resolve(projectDir, '..', '..', 'node_modules', pkg)
+        if (fs.existsSync(monoRepoPath)) {
+            return monoRepoPath
+        }
+    }
+    throw new Error(`Could not find dependency ${pkg}`)
```

Comment:
You mentioned a few dependencies that don't seem to exist anywhere. One of them is `react-query`. It actually can be found in `pwa-kit-react-sdk` package, but not the `pwa-kit-dev`.

Again, I think there's another bug with `sdkDir`. It equates 'sdk' to mean `pwa-kit-dev`, where in fact we may want to check the other pwa-kit-related packages too. I think this is a regression when we moved to V2 of PWA Kit. In V1, there used to be only pwa-kit-react-sdk, but in V2 we split it into smaller packages.